lib/resty/rediscluster.lua

```diff
@@ -139,30 +154,36 @@ local function try_hosts_slots(self, serv_list)
                                   config.send_timeout or DEFAULT_SEND_TIMEOUT,
                                   config.read_timeout or DEFAULT_READ_TIMEOUT)
 
-        --attempt to connect DEFAULT_MAX_CONNECTION_ATTEMPTS times to redis
+        -- attempt to connect DEFAULT_MAX_CONNECTION_ATTEMPTS times to redis
+        -- Kong currently does not configure 'max_connection_attempts'
         for k = 1, config.max_connection_attempts or DEFAULT_MAX_CONNECTION_ATTEMPTS do
+            -- for each ip:port pair, we try at least once before check total timeout
+            ok, err = redis_client:connect(ip, port, self.config.connect_opts)
+            if ok then break end
+            if err then
```

Comment:
Ok but if `err` is always truthy in this case why make it a condition for `if`  🤔  ?

Reply:
I see your point. We can use it as `if ... else ...` or remove the `if` from `if err`.

The `redis_client:connect` is a wrapper for OpenResty's original `tcpsock:connect`. Use `if err` just in case the wrapper returns `nil, nil`.

So far, the `if err` is harmless.

Reply:
Ok I understand that you just moved this code from somewhere else - so let's just minimize the amount of changes for this refactor and take a step by step approach 👍

Reply:
Frankly, I think there's a bug here but it's not like it was introduced by you so maybe I wouldn't say this code here is _harmless_ but it's a bit better than it was.